frontend_tests/puppeteer_tests/04-subscriptions.js

```diff
@@ -0,0 +1,150 @@
+const common = require('../puppeteer_lib/common');
+const assert = require("assert").strict;
+
+async function user_checkbox(page, name) {
+    const user_id = await common.get_user_id_from_name(page, name);
+    return `#user-checkboxes [data-user-id="${user_id}"]`;
+}
+
+async function user_span(page, name) {
+    return await user_checkbox(page, name) + ' input ~ span';
+}
+
+async function stream_checkbox(page, stream_name) {
+    const stream_id = await common.get_stream_id(page, stream_name);
+    return `#stream-checkboxes [data-stream-id="${stream_id}"]`;
+}
+
+async function stream_span(page, stream_name) {
+    return await stream_checkbox(page, stream_name) + ' input ~ span';
+}
+
+async function wait_for_checked(page, user_name, is_checked) {
+    const selector = await user_checkbox(page, user_name);
+    await page.waitForFunction((selector, is_checked) => $(selector).find('input')[0].checked === is_checked,
+                               {}, selector, is_checked);
+}
+
+async function stream_name_error(page) {
+    await page.waitForSelector('#stream_name_error', {visible: true});
+    return await common.get_text_from_selector(page, '#stream_name_error');
+}
+async function subscriptions_tests(page) {
+    await common.log_in(page);
+
+    const all_streams_selector = 'a[href="#streams/all"]';
+    await page.waitForSelector(all_streams_selector, {visible: true});
+    await page.click(all_streams_selector);
+
+    await page.waitForSelector('#subscription_overlay.new-style', {visible: true});
+    assert(page.url().includes('#streams/all'));
+
+    // Test subscribe and unsubscribe button for #Verona
+
+    const verona_sub_unsub_checked_selector = "[data-stream-name='Verona'] .sub_unsub_button.checked";
+    const verona_sub_unsub_unchecked_selector = "[data-stream-name='Verona'] .sub_unsub_button:not(.checked)";
+
+    // assert it's already checked.
+    await page.waitForSelector(verona_sub_unsub_checked_selector, {visible: true});
+    // get subscribe/unsubscribe button emlement.
+    const sub_unsub_element = await page.$("[data-stream-name='Verona'] .sub_unsub_button");
```

Comment:
Maybe `subscription_checkmark`  or `subscription_button` here .

Reply:
`subscription_checkmark` sounds good. Changed it.